### ISSUE TITLE
[Bug] "Set color per face" task panel: color swatch does not update

### ISSUE THREAD

**Issue by Roy-043** [FreeCAD/FreeCAD#6450]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=3&t=66619

### Version

0.20 (Development)

### Full version info

```shell
OS: Windows 8.1 (6.3)
Word size of FreeCAD: 64-bit
Version: 0.20.27518 (Git)
Build type: Release
Python version: 3.8.12
Qt version: 5.12.9
Coin version: 4.0.0
OCC version: 7.5.3
Locale: Dutch/Netherlands (nl_NL)
```

### Subproject(s) affected?

Core

### Issue description

In V0.20 the color swatch in the "Set color per face" task panel does not update when you select a colored face. This does happen in V0.19.
![color_per_face_color_task_panel_issue](https://user-images.githubusercontent.com/70520633/155689718-1723712b-a5db-497b-9cb1-588fed49f9ab.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by EuSouDeus** [comment]
Hum.. I'm trying to recreate it on the 27422 (Git) and it's working well but only applies the color change after i confirm and deselect the highlighted area.

Could you please update to the latest Git version and check if it still doing the same?!

**Comment by EuSouDeus** [comment]
At least on my side is working... the only thing different from yours that i noticed is that you're running windows 8.1 (6.3) while I'm running Windows 10.. not sure but it might be an issue OS related.. unfortunately I only have this laptop don't one another PC with 8.1 on it. Let's hope someone else with the 8.1 version avaliable can give it a try and check if the same issue shows up!

![Sem Título](https://user-images.githubusercontent.com/20737382/156813258-355637a6-5844-4b65-8e10-d160524899c3.png)
0

OS: Windows 10 (10.0)
Word size of FreeCAD: 64-bit
Version: 0.20.27809 (Git)
Build type: Release
Python 3.8.12, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: Portuguese/Portugal (pt_PT)

**Comment by Roy-043** [comment]
You misunderstand the issue. Note that the issue was already confirmed on the forum.

See the attached image for the V0.19 behavior.
![color-swatch-V0 19](https://user-images.githubusercontent.com/70520633/156827610-5a4b408b-bbed-432e-a2b8-be731bf7894a.png)

**Comment by wwmayer** [comment]
@Roy-043 I have tested it with current master (cfe94b00d343) and for me it works as expected. Does it still fail for you?

**Comment by Roy-043** [comment]
This has indeed been fixed now. This issue can be closed. Thanks.
```
OS: Windows 8.1 (6.3)
Word size of FreeCAD: 64-bit
Version: 0.20.28671 (Git)
Build type: Release
Python 3.8.13, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: Dutch/Netherlands (nl_NL)
Installed mods: 
```

**Comment by wwmayer** [comment]
Was fixed with d355b9d9e2f836
